java-manta-cli/src/main/java/com/joyent/manta/cli/MantaCLI.java

```diff
@@ -264,33 +270,47 @@ public void run() {
                          header = "Performs a download of file in Manta",
                          description = "Performs a download of file in Manta.")
     public static class GetFile extends MantaSubCommand {
+        @CommandLine.Option(names = {"-o", "--output"},
+                            description = "write output to <file> instead of stdout")
+        private String outputFileName;
+        @CommandLine.Option(names = {"-O", "--remote-name"},
+                            description = "write output to a file using remote object"
+                            + "name as filename")
+        private boolean inferOutputFileName = false;
+
         @CommandLine.Parameters(index = "0", description = "Object path in Manta to download")
         private String filePath;
 
         @Override
         public void run() {
-            final StringBuilder b = new StringBuilder();
-
-            b.append("Creating connection configuration").append(BR);
             ConfigContext config = buildConfig();
-            b.append(INDENT).append(ConfigContext.toString(config)).append(BR);
-
-            b.append("Creating new connection object").append(BR);
             try (MantaClient client = new MantaClient(config)) {
-                b.append(INDENT).append(client).append(BR);
+                OutputStream out = System.out;
+                if (inferOutputFileName) {
+                    outputFileName = MantaUtils.lastItemInPath(filePath);
+                }
+                if (outputFileName != null) {
+                    out = new FileOutputStream(outputFileName);
+                }
+                BufferedOutputStream bout = new BufferedOutputStream(out, 1024);
 
-                b.append("Attempting GET request to: ").append(filePath).append(BR);
-                b.append("\nMetadata values: \n").append(client.get(filePath).getMetadata().toString());
-                b.append("\n\nPayload: \n");
+                InputStream objectStream = client.getAsInputStream(filePath);
+                int oneByte;
+                while ((oneByte = objectStream.read()) != -1) {
```

Comment:
Any particular reason to reimplement the `IOUtils.copy` loop for a single byte at a time?

Reply:
Accident of development process, no good reason.